src/components/map/Map.tsx

```diff
@@ -157,138 +107,219 @@ export const Map = ({
+        coordinates: [location.position.longitude, location.position.latitude],
+      },
+    }));
+
+    index.load(points);
+    const bounds = [-180, -85, 180, 85];
+    setClusters(index.getClusters(bounds, zoomLevel));
+
+    // Berechne die Begrenzungskoordinaten für alle Marker
+    if (locations.length > 0 && isInitialFit) {
+      const coordinates = locations.map(loc => [loc.position.longitude, loc.position.latitude]);
+      if (locations.length === 1) {
+        if (cameraRef.current) {
+          cameraRef.current.setCamera({
+            centerCoordinate: coordinates[0],
+            zoomLevel: 15,
+            animationDuration: 0,
+          });
+          setTimeout(() => {
+            setIsLoading(false);
+          }, 4000); // Timeout in Millisekunden anpassen je nach Bedarf
+          setIsInitialFit(false); // Verhindert zukünftige automatische Anpassungen
+        }
+      } else {
+        const [minLng, minLat] = coordinates.reduce((prev, curr) => [Math.min(prev[0], curr[0]), Math.min(prev[1], curr[1])], [Infinity, Infinity]);
+        const [maxLng, maxLat] = coordinates.reduce((prev, curr) => [Math.max(prev[0], curr[0]), Math.max(prev[1], curr[1])], [-Infinity, -Infinity]);
+        const deltaLng = (maxLng - minLng) * 0.1; // Add 10% padding to the bounds
+        const deltaLat = (maxLat - minLat) * 0.1; // Add 10% padding to the bounds
+        if (cameraRef.current) {
+          cameraRef.current.fitBounds(
+            [minLng - deltaLng, minLat - deltaLat],
+            [maxLng + deltaLng, maxLat + deltaLat],
+            0 // animationDuration in ms
+          );
+          setTimeout(() => {
+            setIsLoading(false);
+          }, 1000); // Timeout in Millisekunden anpassen je nach Bedarf
+          setIsInitialFit(false); // Verhindert zukünftige automatische Anpassungen
+        }
+      }
+    }
+  }, [locations, clusterDistance, zoomLevel, isInitialFit]);
+
+  const handleMyLocationButtonPress = () => {
+    if (refForMapView.current && locationSettings?.currentLocation) {
+      cameraRef.current.setCamera({
+        centerCoordinate: [
+          locationSettings.currentLocation.latitude,
+          locationSettings.currentLocation.longitude
+        ],
+        zoomLevel: 15,
+        animationDuration: 1000
+      });
+    }
+  };
+
   return (
     <View style={[styles.container, style]}>
-      <MapView
-        clusterColor={colors.primary}
-        clusterFontFamily="regular"
-        clusteringEnabled={clusteringEnabled}
-        radius={clusterDistance}
-        renderCluster={renderCluster}
-        initialRegion={initialRegion}
-        region={updatedRegion}
-        mapType={device.platform === 'android' ? MAP_TYPES.NONE : MAP_TYPES.STANDARD}
+      <LoadingSpinner loading={isLoading} />
+      <MapLibreGL.MapView 
+        style={styles.map}
+        styleJSON='https://tileserver-gl.smart-village.app/styles/osm-liberty/style.json'
         onPress={onMapPress}
```

Comment:
Does the `{nativeEvent}` return we get with `onPress` have the same values as the `onPress` we used before?

Reply:
yes, adjusted in the newest version https://github.com/smart-village-solutions/smart-village-app-app/pull/1154/commits/7d15a3d5f2af482f630518fdc922da1732420e58